sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -214,8 +221,13 @@ public void recordGatewayResponse(
             this.recordRetryContextEndTime();
 
             if (locationEndPoint != null) {
-                this.regionsContacted.add(globalEndpointManager.getRegionName(locationEndPoint, rxDocumentServiceRequest.getOperationType()));
+
+                String regionName = globalEndpointManager.getRegionName(locationEndPoint, rxDocumentServiceRequest.getOperationType());
+
+                this.regionsContacted.add(regionName);
                 this.locationEndpointsContacted.add(locationEndPoint);
+
+                this.regionsContactedWithContext.add(new RegionWithContext(regionName, locationEndPoint));
```

Comment:
Curious, can we combine the this.regionsContact and this. this.regionsContactedWithContext